src/H5FDcore.c

```diff
@@ -190,9 +190,9 @@ static const H5FD_class_t H5FD_core_g = {
 };
 
 /* Default configurations, if none provided */
-static const H5FD_core_fapl_t H5FD_core_default_config_g       = {H5_MB, TRUE, H5FD_CORE_WRITE_TRACKING_FLAG,
+static const H5FD_core_fapl_t H5FD_core_default_config_g       = {(size_t)H5_MB, TRUE, H5FD_CORE_WRITE_TRACKING_FLAG,
                                                             H5FD_CORE_WRITE_TRACKING_PAGE_SIZE};
-static const H5FD_core_fapl_t H5FD_core_default_paged_config_g = {H5_MB, TRUE, TRUE, (size_t)4096};
+static const H5FD_core_fapl_t H5FD_core_default_paged_config_g = {(size_t)H5_MB, TRUE, TRUE, (size_t)4096};
```

Comment:
This is fine, I guess, but I would probably create a const int H5_CORE_DEFAULT_ALLOC_SIZE or something like that and make it 1024 * 1024 instead. That's more in the spirit of what's intended and doesn't torture the floating-point H5_MB into a new use.

Also, the size_t cast for 4096 is probably superfluous.

Reply:
I tried that already and it generated more warnings.

Reply:
Compilers are weird